### ISSUE TITLE
validating schema with `additionalProperties: false`

### ISSUE THREAD

**Issue by punkish** [fastify/help#713]
#### You have already researched for similar issues?

there is a [similar issue](https://github.com/fastify/fastify/issues/3426) to my problem but the fastify version is different and the resolution doesn't work for me. I asked this [question on SO](https://stackoverflow.com/questions/72875186/how-to-remove-non-schema-params-from-the-query) as well, but the answer didn't help.

#### What are you trying to achieve, or the steps to reproduce?

I have the following schema

```js
{
  '$schema': 'https://json-schema.org/draft/2020-12/schema',
  '$id': 'https://example.com/treatments.schema.json',
  title: 'Treatments',
  summary: 'Fetches treatments',
  description: 'Treatments are…',
  response: {},
  querystring: {
    type: 'object',
    additionalProperties: false,
    properties: {
      treatmentId: {
          type: "string",
          maxLength: 32,
          minLength: 32,
          description: "A 32 character string: treatmentId=388D179E0D564775C3925A5B93C1C407"
      },
      … <snipped> …
    }
  },
  tags: [ 'zenodeo' ]
}
```

#### What was the result you received?

I know the schema is being validated because `http://localhost:3010/v3/treatments?treatmentId=foo` gives an error `{"statusCode":400,"error":"Bad Request","message":"querystring/treatmentId must NOT have fewer than 32 characters"}` (that is good 👍🏽) but the `additionalProperties: false` constraint is still not working because `http://localhost:3010/v3/treatments?foo=bar` gives no error and `foo` is also not removed from `request.query` (that is not good 👎🏽)

#### What did you expect?

I expect either `foo=bar` be removed by the validator, or, optionally, if I ask it so, it should croak with an error.

#### Context

* *node version*: 16+
* *fastify version*: >=4.2

**Comment by mcollina** [comment]
Fastify uses Ajv internally. Have you verified that your schema works well with Ajv? 

We have a few default settings that should cover this:

```js
{
  coerceTypes: true, // change data type of data to match type keyword
  useDefaults: true, // replace missing properties and items with the values from corresponding default keyword
  removeAdditional: true, // remove additional properties
  // Explicitly set allErrors to `false`.
  // When set to `true`, a DoS attack is possible.
  allErrors: false
}
```

**Comment by mcollina** [comment]
Could you add a minimum viable reproduction for Fastify? Thanks!

**Comment by punkish** [comment]
@mcollina while trying to create a min viable reproduction of the error, I discovered that *I* was the source of the error. Thank you for pushing me in the right direction, and apologies for the noise. I am closing this issue, and getting back to more careful work.

***note:** I added `removeAdditional: false` to make sure Ajv appropriately croaks so the user is unable to send unspecified query params.*
